tests/apis.js

```diff
@@ -36,96 +35,316 @@ test.cb.before(t => {
-    t.true(itm.hasOwnProperty('type'), 'Contains Type');
-    t.true(itm.hasOwnProperty('type_slug'), 'Contains Type Slug');
-    t.true(itm.hasOwnProperty('key'), 'Contains Key');
-    t.true(itm.hasOwnProperty('key_slug'), 'Contains Key Slug');
-    t.false(itm.hasOwnProperty('meta'), 'Does not contain Meta');
-    t.true(itm.hasOwnProperty('attributes'), 'Contains attributes');
+  return formatted.then(result => {
+    result.forEach(itm => {
+      utils.attribute(t, itm);
+    });
+  });
+});
+
+test.serial.skip('Utils: Format Results - query depth zero', t => {
+  const query = {
+    depth: 0,
+  };
+  const item = Math.round(Math.random() * (live.length - 1));
+  let expected = cloneDeep(live[item]);
+
+  if (expected === undefined) {
+    expected = cloneDeep(live[(live.length - 1)]);
+  }
+
+  const model = allTypes.find(typ => {
+    return typ.id === expected['type-slug'];
+  });
+
+  const formatted = apiUtils.format([expected], model.attributes, allTypes, query);
+
+  return formatted.then(result => {
+    result.forEach(itm => {
+      utils.attribute(t, itm);
+    });
+  });
+});
+
+test.serial.skip('Utils: Format Results - query depth one', t => {
+  const query = {
+    depth: 1,
+  };
+  const item = Math.round(Math.random() * (live.length - 1));
+  let expected = cloneDeep(live[item]);
+
+  if (expected === undefined) {
+    expected = cloneDeep(live[(live.length - 1)]);
+  }
+
+  const model = allTypes.find(typ => {
+    return typ.id === expected['type-slug'];
+  });
+
+  const formatted = apiUtils.format([expected], model.attributes, allTypes, query);
+
+  return formatted.then(result => {
+    result.forEach(itm => {
+      utils.attribute(t, itm);
+      t.true(itm.hasOwnProperty('attributes'), 'Contains attributes');
+
+      Object.keys(itm.attributes).forEach(attr => {
+        if (attr.split('-').indexOf('referencer') > -1) {
+          if (itm.attributes[attr].hasOwnProperty('id')) {
+            utils.attribute(t, itm.attributes[attr]);
+          }
+          else if (!Array.isArray(itm.attributes[attr])) {
+            Object.keys(itm.attributes[attr]).forEach(atr => {
+              utils.attribute(t, itm.attributes[attr][atr]);
+            });
+          }
+        }
+      });
+    });
+  });
+});
+
+test.serial.skip('Utils: Format Results - query depth two', t => {
+  const query = {
+    depth: 2,
+  };
+  const item = Math.round(Math.random() * (live.length - 1));
+  let expected = cloneDeep(live[item]);
+
+  if (expected === undefined) {
+    expected = cloneDeep(live[(live.length - 1)]);
+  }
+
+  const model = allTypes.find(typ => {
+    return typ.id === expected['type-slug'];
+  });
+
+  const formatted = apiUtils.format([expected], model.attributes, allTypes, query);
+
+  return formatted.then(result => {
+    result.forEach(itm => {
+      utils.attribute(t, itm);
+      t.true(itm.hasOwnProperty('attributes'), 'Contains attributes');
+
+      Object.keys(itm.attributes).forEach(attr => {
```

Comment:
This, and the other `depth=1` version of this, feel like it can be split in to a recursive helper function that can be called as many times as we'd like instead of hard-coding this as we are